documentation/workshop1-overview.md

```diff
@@ -0,0 +1,168 @@
+
+## Session 3: CloudFront
+**Duration:** 2 hours  
+**Module Covered:** CloudFront
+
+### Hour 1: Self-Paced Learning Lab
+Students will:
+- Learn what CloudFront is and why it's beneficial
+- Create a CloudFront distribution
+- Connect CloudFront to their S3 bucket
+- Configure caching and compression settings
+- Access their website via CloudFront URL
+
+### Hour 2: Live Walkthrough & Deep Dive
+Instructor will demonstrate:
+- CloudFront distribution creation step-by-step
+- How CDNs work and improve performance
+- Edge locations and global content delivery
+- HTTPS and security benefits
+- Cache behaviors and optimization
+- The complete architecture: S3 + CloudFront
+
+**Key Concepts:** CDN, CloudFront, edge locations, distributions, caching, HTTPS, compression
+
+## Session 4: Introduction to CI/CD
+**Duration:** 2 hours  
+**Module Covered:** Introduction to CI/CD
+
+### Hour 1: Self-Paced Learning Lab
+Students will:
+- Learn CI/CD fundamentals
+- Understand GitHub Actions basics
+- Set up AWS credentials as GitHub secrets
+- Create a deployment workflow file
+- Test automated deployments
+- Troubleshoot common issues
+
+### Hour 2: Live Walkthrough & Deep Dive
+Instructor will demonstrate:
+- CI/CD concepts and benefits
+- GitHub Actions workflow syntax
+- Setting up secrets securely
+- Building the complete deployment pipeline
+- Testing the automation
+- Best practices and troubleshooting tips
+
+**Key Concepts:** CI/CD, GitHub Actions, workflows, automated deployment, secrets management
+
+## Post-Workshop Module: Workshop Wrap-up & Review
+**Format:** Self-paced review material  
+**When:** Need to discuss if this will be done independently or as a part of the final session. 
+
+### Content Includes:
+
+#### 1. What You've Accomplished
+- Recap of the complete architecture built
+- Summary of key technologies mastered
+- Visual diagram of the deployment pipeline
+
+#### 2. Key Concepts Review
+- Links to all module glossaries:
+  - Cloud Fundamentals Glossary
+  - S3 Glossary
+  - CloudFront Glossary
+  - CI/CD Glossary
+- Consolidated list of must-know terms
+
+#### 3. Interview Preparation Questions
+Interactive self-assessment with questions like:
+- "Explain what an S3 bucket is and how it differs from traditional file storage"
+- "What benefits does CloudFront provide that S3 alone does not?"
+- "Describe the CI/CD pipeline you built and why automation is important"
+- "What happens when you push code to your GitHub repository?"
+- "Explain the difference between IaaS, PaaS, and SaaS with examples"
+
+**Delivery Options:**
+- Questions with expandable answers for self-checking
+- Prompt template for AI practice: "I'm preparing for a job interview. Ask me about [topic] and evaluate my answer"
+
+#### 4. Clean Up Your Resources
+Step-by-step instructions to delete:
+- CloudFront distribution
+- S3 bucket and all objects
+- GitHub secrets (optional)
```

Comment:
Should we have the students delete the IAM user we create for github secrets credentials or are we going to do that since IAM wasn't in the curriculum?